### ISSUE TITLE
error during prune: tree xxx not found in repository

### ISSUE THREAD

**Issue by fbartels** [restic/restic#1626]
## Output of `restic version`
```
restic 0.8.2
compiled with go1.10 on linux/amd64
```

## How did you run restic exactly?
I'm storing restic backups on S3/Minio, utilizing https://github.com/binarybucks/restic-tools as a wrapper.

```
$ sudo backup minio rebuild-index && sudo backup minio check && sudo backup minio prune
password is correct
counting files in repo
[0:19] 100.00%  8769 / 8769 packs
finding old index files
saved new indexes as [5133c0ff 2f8bc2d0 64769951]
remove 3 old index files
password is correct
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
error for tree bb4ac29d:
  tree bb4ac29d3c4ba7bee745f248fb0f890d4fd9247a1d6e8d7f6a1a14d512d193a3 not found in repository

Fatal: repository contains errors

$ sudo backup minio prune
password is correct
counting files in repo
building new index for repo
[0:20] 100.00%  8769 / 8769 packs
repository contains 8769 packs (739630 blobs) with 38.938 GiB
processed 739630 blobs: 2 duplicate blobs, 1.938 KiB duplicate
load all snapshots
find data that is still in use for 56 snapshots
tree bb4ac29d3c4ba7bee745f248fb0f890d4fd9247a1d6e8d7f6a1a14d512d193a3 not found in repository
github.com/restic/restic/internal/repository.(*Repository).LoadTree
        /tmp/restic-build-354782844/src/github.com/restic/restic/internal/repository/repository.go:615
github.com/restic/restic/internal/restic.FindUsedBlobs
        /tmp/restic-build-354782844/src/github.com/restic/restic/internal/restic/find.go:11
main.pruneRepository
        src/github.com/restic/restic/cmd/restic/cmd_prune.go:191
main.runPrune
        src/github.com/restic/restic/cmd/restic/cmd_prune.go:85
main.glob..func16
        src/github.com/restic/restic/cmd/restic/cmd_prune.go:25
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).execute
        /tmp/restic-build-354782844/src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:698
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /tmp/restic-build-354782844/src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:783
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).Execute
        /tmp/restic-build-354782844/src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:736
main.main
        src/github.com/restic/restic/cmd/restic/main.go:69
runtime.main
        /usr/local/go/src/runtime/proc.go:198
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:2361
```

## What backend/server/service did you use to store the repository?

S3, a locally setup Minio

## Expected behavior

Run the prune command

## Actual behavior

Stop with a backtrace

## Steps to reproduce the behavior

I do see the error on my setup, but I cannot say how it got into this state.

## Do you have any idea what may have caused this?

No

## Do you have an idea how to solve the issue?

No

## Did restic help you or made you happy in any way?

Apart from the error during prune, restic currently does daily backups from three different hosts.

**Comment by fd0** [comment]
Thanks for creating a separate issue. I don't have any idea yet on how this could have happened. Is there a chance that a file could have been removed from the server? Maybe accidentally? Or did you move the repo from somewhere to that server?

Which versions of restic were used? Can you remember?

Were there any errors reported before?

Usually, restic tries very hard to only write snapshots to the repo if all data could be saved successfully. This should include the index, and since #1589 it also does that for all indexes.

**Comment by fbartels** [comment]
hmm.. before updating to 0.8.2 I've only used 0.8.1. I did have some locking issues which I needed to manually release. A few weeks ago I moved the storage of my Minio from one disk to another (while Minio was shut down, via rsync), but I cannot remember of the error occurred before or only after.

**Comment by fbartels** [comment]
Hi, 

just updated to 0.8.3 and the error seems to have disappeared:

```
$ restic version
restic 0.8.3
compiled with go1.10 on linux/amd64
$ sudo backup minio prune
password is correct
counting files in repo
building new index for repo
[1:48] 100.00%  10676 / 10676 packs
repository contains 10676 packs (812397 blobs) with 47.191 GiB
processed 812397 blobs: 12 duplicate blobs, 10.406 MiB duplicate
load all snapshots
find data that is still in use for 64 snapshots
[1:50] 100.00%  64 / 64 snapshots
found 547748 of 812397 data blobs still in use, removing 264649 blobs
will remove 0 invalid files
will delete 2920 packs and rewrite 1520 packs, this frees 15.739 GiB
[10:01] 100.00%  1520 / 1520 packs rewritten
counting files in repo
[1:11] 100.00%  7162 / 7162 packs
finding old index files
saved new indexes as [632f5e8d 7699b46b 7898f5a5]
remove 51 old index files
[0:26] 100.00%  4440 / 4440 packs deleted
done
```

**Comment by fd0** [comment]
Nice, thanks for the feedback. The most likely explanation is that you ran into #1589 but only noticed it recently. That'd fit with that no data was actually lost.

**Comment by kumy** [comment]
@fd0 I'm also having the problem on `0.9.4` would you like a dedicated issue?

```
[root@host ~]# restic -r ${REPO} check
using temporary cache in /tmp/restic-check-cache-501369179
repository 6178799c opened successfully, password is correct
created new cache in /tmp/restic-check-cache-501369179
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
error for tree 98e92b34:
  tree 98e92b3420312dc928674a8f6a37a1814168775bb50f4b1a247dc9c6a70e2022 not found in repository
Fatal: repository contains errors

[root@host ~]# restic -r ${REPO} rebuild-index
repository 6178799c opened successfully, password is correct
counting files in repo
[0:00] 100.00%  43 / 43 packs
finding old index files
saved new indexes as [0e04df7e]
remove 1 old index files

[root@host ~]# restic -r ${REPO} check
using temporary cache in /tmp/restic-check-cache-935211248
repository 6178799c opened successfully, password is correct
created new cache in /tmp/restic-check-cache-935211248
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
error for tree 98e92b34:
  tree 98e92b3420312dc928674a8f6a37a1814168775bb50f4b1a247dc9c6a70e2022 not found in repository
Fatal: repository contains errors

[root@host ~]# restic version
restic 0.9.4 compiled with go1.11.4 on linux/amd64
```
